docs/sdk/src/guides/enable_elastic_scaling.rs

```diff
@@ -0,0 +1,249 @@
+//! Below are described the constraints of the implementation:
+//!
+//! 1. **Bounded compute throughput**. Each parachain block gets at most 2 seconds of execution on
+//!    the relay chain. Therefore, assuming the full 2 seconds are used, a parachain can only
+//!    utilise at most 3 cores in a relay chain slot of 6 seconds. If the full execution time is not
+//!    being used or if collators are able to author blocks faster than the validators are able to
+//!    run them, higher core counts can be achieved.
+//! 2. **Sequential block execution**. Each collator must import the previous block before authoring
+//!    a new one. At present this happens sequentially, which limits the maximum compute throughput when
+//!    using multiple collators. To briefly explain the reasoning: first, the previous collator spends
+//!    2 seconds building the block and announces it. The next collator fetches and executes it, wasting
+//!    2 seconds plus the block fetching duration out of its 2 second slot. Therefore, the next collator
+//!    cannot build a subsequent block in due time and ends up authoring a fork, which defeats the purpose
+//!    of elastic scaling. The highest throughput can therefore be achieved with a single collator but
+//!    this should obviously only be used for testing purposes, due to the clear lack of decentralisation
+//!    and resilience. Experiments show that the peak compute throughput using more than one collator
+//!    (measured up to 10 collators) is utilising 2 cores with authorship time of 1.3 seconds per block,
+//!    which leaves 400ms for networking overhead. This would allow for 2.6 seconds of execution, compared
+//!    to the 2 seconds async backing enabled. If block authoring duration is low and you attempt to
+//!    use elastic scaling for achieving low latency or increasing storage throughput, this is not a
+//!    problem. Developments required for streamlining block production are tracked by [this issue](https://github.com/paritytech/polkadot-sdk/issues/5190).
+//! 3. **Lack of out-of-the-box automated scaling.** For true elasticity, the parachain must be able
+//!    to seamlessly acquire or sell coretime as the user demand grows and shrinks over time, in an
+//!    automated manner. This is
+//!    currently lacking - a parachain can only scale up or down by “manually” acquiring coretime.
+//!    This is not in the scope of the relay chain functionality. Parachains can already start
+//!    implementing such autoscaling, but we aim to provide a framework/examples for developing
+//!    autoscaling strategies.
+//!    Tracked by [this issue](https://github.com/paritytech/polkadot-sdk/issues/1487).
+//!    An in-progress external implementation by RegionX can be found [here](https://github.com/RegionX-Labs/On-Demand).
+//!
+//! Another important constraint is that when a parachain forks, the number of blocks backed per
+//! relay chain block decreases.
```

Comment:
It is much easier for someone to understand if we mention clearly the impact first and the technical reason after.
```suggestion
//! Another important constraint is that when a parachain forks, the throughput decreases and latency increases because the number of blocks backed per
//! relay chain block decreases.
```